### DISCUSSION TITLE
Standalone Classmap Generator

### DISCUSSION THREAD

**Discussion by barryvdh** [composer/composer#10527] [General]
The Classmap Generator used in Composer is based on https://symfony.com/doc/3.3/components/class_loader/class_map_generator.html
That component is not developed anymore and outdated, the refers back to Composer.

If packages want to use the Classmap Generator, they are forced to require composer/composer in its entirity.

Based on https://twitter.com/seldaek/status/1491125412424224768 you suggested to make a standalone version, which I would appreciate (at least for https://github.com/barryvdh/laravel-ide-helper to scan for Models).

The component is relatively standalone, mostly just requires composer/pcre, symfony/finder and a few methods from the Filesystem class.

I would be happy to help out with this if needed.

**Comment by Seldaek** [comment]
I'll try to do this next week, probably can get rid of the Util\Filesystem dependencies, and I will probably take the opportunity to clean up the API a little bit. Anyway I guess your copy works for now, so no rush.

**Reply by Seldaek** [reply]
Are we next week already? 😄 

Oh well, a bit late but here it is slowly taking shape https://github.com/composer/class-map-generator - still need to refactor a couple things and document it but feel free to have a look already.

**Reply by Seldaek** [reply]
OK 1.0.0 has been tagged on https://github.com/composer/class-map-generator and integration in Composer is going well https://github.com/composer/composer/pull/10885 👍🏻 

cc @barryvdh @mfn @canvural @spawnia I'd be happy if you can get all the dependents to use that package instead of copy-pasted code or especially instead of `composer/composer` requires! Thanks :)

**Reply by barryvdh** [reply]
Thanks, looks good so far :) https://github.com/barryvdh/laravel-ide-helper/pull/1357

**Reply by canvural** [reply]
It works great. Thank you!

**Comment by barryvdh** [comment]
Yeah definitely no rush from my side :)

**Comment by mfn** [comment]
I guess https://github.com/nunomaduro/larastan/pull/1119 also joined the waiting list 😄

**Comment by spawnia** [comment]
This also affects https://github.com/BenSampo/laravel-enum/issues/249
